### DISCUSSION TITLE
umi3迁移umi4中，img导入本地图片不显示

### DISCUSSION THREAD

**Discussion by haocan** [umijs/umi#11743] [Q&A]
umi3迁移umi4中页面img使用以下方法导入本地图片：
<img src={require('@/assets/images/new_order/车辆基本信息icon@2x.png')} />
umi3和umi4都把图片转换成了base64但不一样 ，umi3转换结果：data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAEwAAABMCAYAAADHl1ErAAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAATKADAAQAAAABAAAATAAAAAAWucfgAAAF60lEQVR4Ae2cf2gcRRTHv2/vmoa2JrFVm6r1R6nxtwX1D0GtRBGjIApqsDEGfwSxNklRQZEWGgQVqqghKpYIYkIrVKglqGAQ/1BE0Fr8UVpNRSKmTVNNKyW/Lne3z7fXbC57u3e7k1uT3GUGyuy8ee/NzGdn383OTgropAloApqAJqAJaALeBChTfOEOXhUbQxsId4BRlllfLGUZ+ISM5aBhYOuRFvo06LhcwCrb+EsBVR3UQRHoJaKEK/o30+EgYzE8lNZ7yIpZFGXCjUEH6AbGiAQ1LhY9BtwcsgwusGIW+wUn1sAUb7kGpoEpElBU1zNMA1MkoKiuZ5gGpkhAUV3PMA1MkYCiejSoPhF+YsZQUP35qCc7DTfJa1BJPn1TAfbcwGbqyaexubZd9Sb3Sx/Oy6cfOoYp0tPANDBFAorqeoZpYIoEFNX1DNPAFAkoqgdeh/n5XbODy8dieFBlf9zPp1+9LEQPDLTQ1356dr187CD5IpZXCg3YaAz3wMS7efVG1ZhwUEyuDGpGDHlZyS+FFsPEUWjwAw9pDr5whQYs8CALXHFhASOMet4vwoin3EMY3mMk8cHD//8tMlUaiCzFDeYYzpluI7sw5pEm6p0uy3UdGjCT8IkE1S3yDXnWZq0M9udcg8us62+kEyKz/s04hQbsWAv9Lb14ecY9KRDDWZsNBcLDt5samC8ip4IG5uThW9LAfBE5FUIL+ue/zWsTCbzgdK9Wkrs3voTwzO8tFFOznD3t0ICZSayXldhj+XTdWlSNReR8LRB4XWS1x7saVyIx2iHt3y6lUkuWO5E0xcdBdFRs9oO4G6WVX1DtG2O57RRO3vk5CqueDdlRUE3xkbfktfruYLAs59IKUCk218p1o0DrxtixP7mzroV3t+b8DFcsMewWVcYe+mcLwDaM//Yj76y7xKM+JSoOYESLsw1QWc64HAl8x50bPG9CaMDkRTKUQF1C4fhRBuUw4Aow7eWu+ksdYimEFvQjldiTHESNbCKqx6DJXkUMjPdtpL7MTs6oTMYT1LCzI5cts+zB7npoLUy+VQC9JI/kirS+BS3ZzZ81X0N3tU9NhtCA9deS9QvzebrBOb4y0O/XAyLZLgCsP2g4LL+0exEf+V6uV0/ZMaowNNQs5ddsWWiPpO2wUHOqe29QlhlPuvpvYgt3PrzUlmtgNgkrX13VIwElYy0mjyYlamw1DcwmITlVtyYkcy+aTdxrq2lgNgk7Z7KgORPhaluggdkkcuWMc+1qDcwmkSsnnGVXh7assB3Om9w075SFZ0YA9+1dKdi8SN4tnYnTa8viBcZoloWntYYKNelHUhGnBqYIrOAeSf6grtU1Rubwditczp2CggMmG37bnENQLBGdlKD+h/iRF226QHKlp6wAgSkCstWJ9gCRV6iha58t4o8fqcDwxFOyW/GsyJbb8lx5YGBs4mb5w4AzppwZrh/fqarUhc9ZC/KzN0/75yh6BzbRIadzhZJ1hI7oadRXtaOrt0k2Bl+XmbVOpEdxauJblCzeBjP2EZKy08J8sZ/n4MCArQ5n8hkhn8TJgNaJ1PmJdQG1vdReRcWyDnT2fiWP3+R/t5BaaJWJ8mWIx+6DQQ+AIzWg+C8CssD39PM5NEd0XPZIX8TJ4e1pWBlMmctgmp1YFLcOqbRn1LqKXgEvz7njamPuBMw9KF9kPfwbc3aCsRJxPAqKSJzzSNNOxrqAyRGiHg+TwhQR7Zc4dV3AX8LrseLMHzwHyqmztKkqVwwzIqhPJrFdNuZvkztzOshbp49zpWnvWl5qqdPLXhVpWVb/UvFrWk31SnF9Nhzz+gh8Qka/yW7ZBUxO4w1J5eO2wrzLjUh14D5Fo31YYgzh33EJM77rrX1YI0c6D03zTzyOkvIDVPvOsN1m1jtrKxRDLm8HEsy5KetYCIPy/ewqqvvwn6w6kxWuGOZnUJD1y5c9L2uvbzz7TnQKhtEQBJZlvyBmmDVQ5lbDWrjK1f1TC1cgtXClDe//ZenopAloAppAsRP4D13DcwFNpse6AAAAAElFTkSuQmCC
umi4转换结果：data:image/png;base64,bW9kdWxlLmV4cG9ydHMgPSAiZGF0YTppbWFnZS9wbmc7YmFzZTY0LGlWQk9SdzBLR2dvQUFBQU5TVWhFVWdBQUFFd0FBQUJNQ0FZQUFBREhsMUVyQUFBQUFYTlNSMElBcnM0YzZRQUFBRVJsV0VsbVRVMEFLZ0FBQUFnQUFZZHBBQVFBQUFBQkFBQUFHZ0FBQUFBQUE2QUJBQU1BQUFBQkFBRUFBS0FDQUFRQUFBQUJBQUFBVEtBREFBUUFBQUFCQUFBQVRBQUFBQUFXdWNmZ0FBQUY2MGxFUVZSNEFlMmNmMmdjUlJUSHYyL3Ztb2EySnJGVm02cjFSNm54dHdYMUQwR3RSQkdqSUFwcXNERUdmd1N4TmtsUlFaRVdHZ1FWcXFnaEtwWUlZa0lyVktnbHFHQVEvMUJFMEZyOFVWcE5SU0ttVFZOTkt5Vy9MbmUzejdmWGJDNTd1M2U3azF1VDNHVUd5dXk4ZWUvTnpHZG4zODNPVGdyb3BBbG9BcHFBSnFBSmFBTGVCQ2hUZk9FT1hoVWJReHNJZDRCUmxsbGZMR1VaK0lTTTVhQmhZT3VSRnZvMDZMaGN3Q3JiK0VzQlZSM1VRUkhvSmFLRUsvbzMwK0VnWXpFOGxOWjd5SXBaRkdYQ2pVRUg2QWJHaUFRMUxoWTlCdHdjc2d3dXNHSVcrd1VuMXNBVWI3a0dwb0VwRWxCVTF6Tk1BMU1rb0tpdVo1Z0dwa2hBVVYzUE1BMU1rWUNpZWpTb1BoRitZc1pRVVAzNXFDYzdEVGZKYTFCSlBuMVRBZmJjd0dicXlhZXh1YlpkOVNiM1N4L095NmNmT29ZcDB0UEFOREJGQW9ycWVvWnBZSW9FRk5YMUROUEFGQWtvcWdkZWgvbjVYYk9EeThkaWVGQmxmOXpQcDErOUxFUVBETFRRMTM1NmRyMTg3Q0Q1SXBaWENnM1lhQXozd01TN2VmVkcxWmh3VUV5dURHcEdESGxaeVMrRkZzUEVVV2p3QXc5cERyNXdoUVlzOENBTFhIRmhBU09NZXQ0dndvaW4zRU1ZM21NazhjSEQvLzh0TWxVYWlDekZEZVlZenBsdUk3c3c1cEVtNnAwdXkzVWRHakNUOElrRTFTM3lEWG5XWnEwTTl1ZGNnOHVzNjIra0V5S3ovczA0aFFic1dBdjlMYjE0ZWNZOUtSRERXWnNOQmNMRHQ1c2FtQzhpcDRJRzV1VGhXOUxBZkJFNUZVSUwrdWUveldzVENiemdkSzlXa3JzM3ZvVHd6Tzh0RkZPem5EM3QwSUNaU2F5WGxkaGorWFRkV2xTTlJlUjhMUkI0WFdTMXg3c2FWeUl4MmlIdDN5NmxVa3VXTzVFMHhjZEJkRlJzOW9PNEc2V1ZYMUR0RzJPNTdSUk8zdms1Q3F1ZURkbFJVRTN4a2Jma3RmcnVZTEFzNTlJS1VDazIxOHAxbzBEcnh0aXhQN216cm9WM3QrYjhERmNzTWV3V1ZjWWUrbWNMd0RhTS8vWWo3Nnk3eEtNK0pTb09ZRVNMc3cxUVdjNjRIQWw4eDUwYlBHOUNhTURrUlRLVVFGMUM0ZmhSQnVVdzRBb3c3ZVd1K2tzZFlpbUVGdlFqbGRpVEhFU05iQ0txeDZESlhrVU1qUGR0cEw3TVRzNm9UTVlUMUxDekk1Y3RzK3pCN25wb0xVeStWUUM5Skkva2lyUytCUzNaelo4MVgwTjN0VTlOaHRDQTlkZVM5UXZ6ZWJyQk9iNHkwTy9YQXlMWkxnQ3NQMmc0TEwrMGV4RWYrVjZ1VjAvWk1hb3dOTlFzNWRkc1dXaVBwTzJ3VUhPcWUyOVFsaGxQdXZwdllndDNQcnpVbG10Z05na3JYMTNWSXdFbFl5MG1qeVlsYW13MURjd21JVGxWdHlZa2N5K2FUZHhycTJsZ05nazdaN0tnT1JQaGFsdWdnZGtrY3VXTWMrMXFEY3dta1Nzbm5HVlhoN2Fzc0IzT205dzA3NVNGWjBZQTkrMWRLZGk4U040dG5ZblRhOHZpQmNab2xvV250WVlLTmVsSFVoR25CcVlJck9BZVNmNmdydFUxUnVid2RpdGN6cDJDZ2dNbUczN2JuRU5RTEJHZGxLRCtoL2lSRjIyNlFIS2xwNndBZ1NrQ3N0V0o5Z0NSVjZpaGE1OHQ0bzhmcWNEd3hGT3lXL0dzeUpiYjhseDVZR0JzNG1iNXc0QXpwcHdacmgvZnFhclVoYzlaQy9Lek4wLzc1eWg2QnpiUklhZHpoWkoxaEk3b2FkUlh0YU9ydDBrMkJsK1htYlZPcEVkeGF1SmJsQ3plQmpQMkVaS3kwOEo4c1ovbjRNQ0FyUTVuOGhraG44VEpnTmFKMVBtSmRRRzF2ZFJlUmNXeURuVDJmaVdQMytSL3Q1QmFhSldKOG1XSXgrNkRRUStBSXpXZytDOENzc0QzOVBNNU5FZDBYUFpJWDhUSjRlMXBXQmxNbWN0Z21wMVlGTGNPcWJSbjFMcUtYZ0V2ejduamFtUHVCTXc5S0Y5a1Bmd2JjM2FDc1JKeFBBcUtTSnp6U05OT3hycUF5UkdpSGcrVHdoUVI3WmM0ZFYzQVg4THJzZUxNSHp3SHlxbXp0S2txVnd3eklxaFBKckZkTnVadmt6dHpPc2hicDQ5enBXbnZXbDVxcWRQTFhoVnBXVmIvVXZGcldrMzFTbkY5Tmh6eitnaDhRa2EveVc3WkJVeE80dzFKNWVPMndyekxqVWgxNEQ1Rm8zMVlZZ3poMzNFSk03N3JyWDFZSTBjNkQwM3pUenlPa3ZJRFZQdk9zTjFtMWp0ckt4UkRMbThIRXN5NUtldFlDSVB5L2V3cXF2dnduNnc2a3hXdUdPWm5VSkQxeTVjOUwydXZieno3VG5RS2h0RVFCSlpsdnlCbW1EVlE1bGJEV3JqSzFmMVRDMWNndFhDbERlLy9aZW5vcEFsb0FwcEFzUlA0RDEzRGN3Rk5wc2U2QUFBQUFFbEZUa1N1UW1DQyI=

**Comment by fz6m** [comment]
在 umi 4 中导入应该均使用 esm ，所以应该直接导入图片获得 url 来使用：

```ts
import imageUrl from '/path/to/img.png'

<img src={imageUrl} alt='image' />
```
